src/mdarray/attribute.jl

```diff
@@ -0,0 +1,254 @@
+    return GDAL.gdalattributegetfullname(attribute)
+end
+
+function gettotalelementscount(attribute::AbstractAttribute)::Int64
+    @assert !isnull(attribute)
+    return Int64(GDAL.gdalattributegettotalelementscount(attribute))
+end
+
+function Base.length(attribute::AbstractAttribute)
+    @assert !isnull(attribute)
+    return Int(gettotalelementscount(attribute))
+end
+
+function getdimensioncount(attribute::AbstractAttribute)::Int
+    @assert !isnull(attribute)
+    return Int(GDAL.gdalattributegetdimensioncount(attribute))
+end
+
+Base.ndims(attribute::AbstractAttribute)::Int = getdimensioncount(attribute)
+
+function getdimensions(
+    attribute::AbstractAttribute,
+)::AbstractVector{<:AbstractDimension}
+    @assert !isnull(attribute)
+    dimensionscountref = Ref{Csize_t}()
+    dimensionshptr =
+        GDAL.gdalattributegetdimensions(attribute, dimensionscountref)
+    dataset = attribute.dataset.value
+    dimensions = AbstractDimension[
+        IDimension(unsafe_load(dimensionshptr, n), dataset) for
+        n in 1:dimensionscountref[]
+    ]
+    GDAL.vsifree(dimensionshptr)
+    return dimensions
+end
+
+function unsafe_getdimensions(
+    attribute::AbstractAttribute,
+)::AbstractVector{<:AbstractDimension}
+    @assert !isnull(attribute)
+    dimensionscountref = Ref{Csize_t}()
+    dimensionshptr =
+        GDAL.gdalattributegetdimensions(attribute, dimensionscountref)
+    dataset = attribute.dataset.value
+    dimensions = AbstractDimension[
+        Dimension(unsafe_load(dimensionshptr, n), dataset) for
+        n in 1:dimensionscountref[]
+    ]
+    GDAL.vsifree(dimensionshptr)
+    return dimensions
+end
+
+function unsafe_getdatatype(
+    attribute::AbstractAttribute,
+)::AbstractExtendedDataType
+    @assert !isnull(attribute)
+    return ExtendedDataType(GDAL.gdalattributegetdatatype(attribute))
+end
+
+function getdatatype(attribute::AbstractAttribute)::AbstractExtendedDataType
+    @assert !isnull(attribute)
+    return IExtendedDataType(GDAL.gdalattributegetdatatype(attribute))
+end
+
+function getblocksize(
+    attribute::AbstractAttribute,
+    options::OptionList = nothing,
+)::AbstractVector{Int64}
+    @assert !isnull(attribute)
+    count = Ref{Csize_t}()
+    blocksizeptr = GDAL.gdalattributegetblocksize(
+        attribute,
+        count,
+        CSLConstListWrapper(options),
+    )
+    blocksize = Int64[unsafe_load(blocksizeptr, n) for n in 1:count[]]
+    GDAL.vsifree(blocksizeptr)
+    return blocksize
+end
+
+function getprocessingchunksize(
+    attribute::AbstractAttribute,
+    maxchunkmemory::Integer,
+)::AbstractVector{Int64}
+    @assert !isnull(attribute)
+    count = Ref{Csize_t}()
+    chunksizeptr = GDAL.gdalattributegetprocessingchunksize(
+        attribute,
+        count,
+        maxchunkmemory,
+    )
+    chunksize = Int64[unsafe_load(chunksizeptr, n) for n in 1:count[]]
+    GDAL.vsifree(chunksizeptr)
+    return chunksize
+end
+
+# processperchunk
```

Comment:
Yes. I added such comments for all functions that I didn't wrap. This function is difficult to wrap because it expects a function pointer, but it's not impossible.

Reply:
> This function is difficult to wrap because it expects a function pointer, but it's not impossible.

Yeah please feel free to leave those in comments like you did

Reply:
Yes it is. I don't know (yet) how to handle callbacks so I didn't wrap this function.